crates/networking/rpc/engine/payload.rs

```diff
@@ -179,7 +179,20 @@ fn parse_execution_payload(params: &Option<Vec<Value>>) -> Result<ExecutionPaylo
     if params.len() != 1 {
         return Err(RpcErr::BadParams("Expected 1 params".to_owned()));
     }
-    serde_json::from_value(params[0].clone()).map_err(|_| RpcErr::WrongParam("payload".to_string()))
+    let payload: ExecutionPayload = serde_json::from_value(params[0].clone())
+        .map_err(|_| RpcErr::WrongParam("payload".to_string()))?;
+
+    match payload {
+        ExecutionPayload {
+            withdrawals: None, ..
```

Comment:
I don't think this is ok to check here, as this code is run for both V1 and V2 🤔